### PR TITLE
Remove explicit mapping of ratings to integers from RBM notebooks

### PR DESCRIPTION
**Pull request by pradnyeshjoshi**
### Description

Previous fix (PR #1581) required user ratings to be explicitly mapped to integers in RBM notebooks. This PR implements changes to the current RBM implementation, so that this explicit mapping is not needed. In particular, a lookup table is maintained internally that maps integer indices to their float rating counterparts.

### Checklist:

- [x] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [x] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

File: recommenders/models/rbm/rbm.py [hunk 1]
```diff
@@ -7,6 +7,8 @@
 import logging
 import time as tm
 
+from tensorflow.python import tf2
```
Comment (miguelgfierro) [inline recommenders/models/rbm/rbm.py:10]: I don't think this is used?

**Review by miguelgfierro — APPROVED** [review]
LGTM

**Comment by anargyri** [comment]
> LGTM

Please don't merge into staging yet

File: recommenders/models/rbm/rbm.py (hunk 1 above)
Comment (pradnyeshjoshi) [reply recommenders/models/rbm/rbm.py]: Removed the import.

**Comment by anargyri** [comment]
> > LGTM
> 
> Please don't merge into staging yet

You may merge now.

**Comment by miguelgfierro** [comment]
everything is green so I'll merge

**Pull request merged**